### PR TITLE
[BUG] PSACT View not displaying Description field correctly 

### LINKED ISSUE #8335: [BUG] PSACT View not displaying Description field correctly
**Rancher Server Setup**
- Rancher version:  `v2.7-62c5e321da5b576e565adf12c666489de9ea9373-head`

**Information about the Cluster**
- Kubernetes version: `v1.25.6+rke2r1`

**User Information**
- What is the role of the user logged in? 
  - Admin

**Describe the bug**

The UI is not correctly showing the field Description for PSACT when in Form mode.

**To Reproduce**

- Open the `rancher-privileged` PSACT (`/dashboard/c/local/explorer/management.cattle.io.podsecurityadmissionconfigurationtemplate/rancher-privileged`).
- Take note of the Description field.
- Switch to YAML view, compare the description field.

**Result**

The description field is populated with: `Any text you want that better describes this resource`
The YAML version description is: `The default privileged pod security admission configuration template`.

**Expected Result**
The value of the description field should be aligned regardless of the view mode.

**Screenshots**
image

image

**Additional context**

### PR DESCRIPTION
**Pull request by cnotv**
### Summary
Fixes #8335

### Occurred changes and/or fixed issues
- Display descriptions in the list
- Allow to edit description from system PSACT

### Technical notes summary
- Allow Steve class to keep `description` key
- Set description as key for the `NameNsDescription` markup of the view

Note: The Rancher server tries to set the description in an analogue field named `description`, while Steve API uses some very specific path for it: `metadata.annotations.field.cattle.io/description`.
Since this is the standard of the model, we'll enforce the path from hardcoded PSACT to this on edit.
We'll still ensure to keep the value.
The description field is always deleted by existing logic in some hidden side-effect.

### Screenshot/Video

https://user-images.githubusercontent.com/5009481/223437657-ae79ee51-d57b-4c1a-9608-35180d7ba20a.mov

### CODE REVIEW

File: shell/plugins/steve/steve-class.js [hunk 1]
```diff
@@ -2,6 +2,19 @@ import { DESCRIPTION } from '@shell/config/labels-annotations';
 import HybridModel from './hybrid-class';
 
 export default class SteveModel extends HybridModel {
+  constructor(data, ctx, rehydrateNamespace = null, setClone = false) {
+    let _description;
```
Comment (richard-cox) [inline shell/plugins/steve/steve-class.js:6]: Haven't looked in to this, but double check SteveDescriptionModel, it might do exactly this

**Review by torchiaf — CHANGES_REQUESTED** [review]
It works fine for `PSACT` in particular, but, shouldn't we find a general solution to this issue? The other resources are still affected by the issue.
For example, just try to set the description of any `Roles` resource.

For what I can see, the API is not updating the `description` field (Trying to update the description for `Administrator` role):
```

PUT request: https://192.168.1.2:8005/v3/globalRoles/admin


Payload:
{
  "annotations": {
    "authz.management.cattle.io/cr-name": "cattle-globalrole-admin",
    "lifecycle.cattle.io/create.mgmt-auth-gr-controller": "true"
  },
  "baseType": "globalRole",
  "builtin": true,
  "created": "2023-01-31T10:28:30Z",
  "createdTS": 1675160910000,
  "creatorId": null,
  "description": "try to change the description",
  "id": "admin",
  "labels": {
    "authz.management.cattle.io/bootstrapping": "default-globalrole",
    "cattle.io/creator": "norman"
  },
  "links": {
    "globalRoleBindings": "https://192.168.1.2:8005/v3/globalRoleBindings?globalRoleId=admin",
    "self": "https://192.168.1.2:8005/v3/globalRoles/admin",
    "update": "https://192.168.1.2:8005/v3/globalRoles/admin"
  },
  "name": "Admin",
  "newUserDefault": false,
  "rules": [
    {
      "apiGroups": [
        "*"
      ],
      "resources": [
        "*"
      ],
      "verbs": [
        "create",
        "delete",
        "get",
        "list",
        "patch",
        "update",
        "watch"
      ]
    },
    {
      "nonResourceURLs": [
        "*"
      ],
      "verbs": [
        "create",
        "delete",
        "get",
        "list",
        "patch",
        "update",
        "watch"
      ]
    }
  ],
  "type": "globalRole",
  "uuid": "db0119cf-3b0a-4801-b838-ffdc9f19a04d"
}


Response:
{
  "annotations": {
    "authz.management.cattle.io/cr-name": "cattle-globalrole-admin",
    "lifecycle.cattle.io/create.mgmt-auth-gr-controller": "true"
  },
  "baseType": "globalRole",
  "builtin": true,
  "created": "2023-01-31T10:28:30Z",
  "createdTS": 1675160910000,
  "creatorId": null,
  "description": "",
  "id": "admin",
  "labels": {
    "authz.management.cattle.io/bootstrapping": "default-globalrole",
    "cattle.io/creator": "norman"
  },
  "links": {
    "globalRoleBindings": "https://192.168.1.2:8005/v3/globalRoleBindings?globalRoleId=admin",
    "self": "https://192.168.1.2:8005/v3/globalRoles/admin",
    "update": "https://192.168.1.2:8005/v3/globalRoles/admin"
  },
  "name": "Admin",
  "newUserDefault": false,
  "rules": [
    {
      "apiGroups": [
        "*"
      ],
      "resources": [
        "*"
      ],
      "type": "/v3/schemas/policyRule",
      "verbs": [
        "*"
      ]
    },
    {
      "nonResourceURLs": [
        "*"
      ],
      "type": "/v3/schemas/policyRule",
      "verbs": [
        "*"
      ]
    }
  ],
  "type": "globalRole",
  "uuid": "db0119cf-3b0a-4801-b838-ffdc9f19a04d"
}

```

The `description` field is empty in the response object.
@cnotv Could you confirm that the root cause of the issue is the missing update from Rest API ?

**Comment by cnotv** [comment]
As I mention, if the model uses that path, we stick to that.
`GlobalRole` uses `SteveDescriptionModel` and the description, it means that they are not working. 
That requires a model update which is out of the scope. I may also not be aware of all the cases needed for this model, so it should also defined in the issue to be tested.

> It works fine for `PSACT` in particular, but, shouldn't we find a general solution to this issue? The other resources are still affected by the issue. For example, just try to set the description of any `Roles` resource.
> 
> For what I can see, this is a back-end issue; the API is not updating the `description` field (Trying to update the description for `Administrator` role):
>
> The `description` field is empty in the response object. @cnotv Could you confirm that the root cause of the issue is the missing update from Rest API ?

It's definitely not a server issue, as we cut it on submit. Check the payload.

This PR is about the PSACT, so we are not going to fix everything. As I mentioned, if the model uses another path as default, we stick to that instead of make exception without know why. To be be fixed we should check case by case.

`GlobalRole` uses `SteveDescriptionModel` and the `description` key, it means that they are not working already. 
That requires a model update which is out of the scope. I may also not be aware of all the cases needed for this model, so it should also defined in the issue to be tested.

Let's stick to the requirement.

**Comment by torchiaf** [comment]
The payload actually contains the `description` to update:

image

`Payload`:
image

`Response`:
image

**Comment by cnotv** [comment]
It seems different for PSA. It does not reach the payload at all even if I use the same class.

https://user-images.githubusercontent.com/5009481/223728806-465bf269-99b6-4480-9654-1367e1b07943.mov

**Comment by richard-cox** [comment]
I've created https://github.com/rancher/dashboard/pull/8377 which should resolve the issue

**Comment by cnotv** [comment]
> I've created #8377 which should resolve the issue

Could you please explain what was the issue and how it has been solved? Also I am still in disagreement to have exception in global configurations using a model without a reason.

**Comment by richard-cox** [comment]
> > I've created #8377 which should resolve the issue
> 
> Could you please explain what was the issue and how it has been solved? Also I am still in disagreement to have exception in global configurations using a model without a reason.

`SteveDescriptionModel` was used by two resources that saved via their norman counterpart rather than steve. This was the first resource type to save by steve and there was a bug (the description, being a `get`ter, wasn't serialised and sent in the http request). The fix is to ensure the model has a chance to tweak the object before sending, which in this case is to serialise itself and manually set `description`

**Comment by richard-cox** [comment]
Closing for https://github.com/rancher/dashboard/pull/8379 / https://github.com/rancher/dashboard/pull/8377

**Comment by richard-cox** [comment]
The class already exists - `SteveDescriptionModel`. We could override the save in there, but it would do exactly what the `resource-class` save does except use a different source object to save with. The solution provide cuts out the duplication and allows other models to use the same approach.

**Pull request closed without merging**